src/providers/authentication/authentication.ts

```diff
@@ -39,8 +61,9 @@ export class AuthenticationProvider {
   }
 
   public login = () => {
-    return new Promise((resolve, reject) => {
-      this.aquireTokenSilently()
+    if (!this.isInUnAuthenticatedMode()) {
```

Comment:
could you invert this if and switch the logic around. It would create a smaller and more readable if

Reply:
Great catch. have done.